x-pack/plugins/triggers_actions_ui/public/application/lib/rule_api/load_action_error_log.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { HttpSetup } from '@kbn/core/public';
+import type { SortOrder } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { escapeKuery } from '@kbn/es-query';
+import { IExecutionErrorsResult, ActionErrorLogSortFields } from '@kbn/alerting-plugin/common';
+import { INTERNAL_BASE_ALERTING_API_PATH } from '../../constants';
+
+export interface SortField {
+  sort_field: ActionErrorLogSortFields;
+  sort_order: SortOrder;
+}
+
+export interface LoadActionErrorLogProps {
+  id: string;
+  runId?: string;
+  message?: string;
+  dateStart: string;
+  dateEnd?: string;
+  filter?: string[];
+  perPage?: number;
+  page?: number;
+  sort?: SortField[];
+}
+
+const SORT_MAP: Record<string, string> = {
+  timestamp: '@timestamp',
+};
+
+// Remaps sorting keys to ones that the server knows how to handle
+const getRenamedSort = (sort?: SortField[]) => {
+  if (!sort) {
+    return [];
+  }
+  return sort.map(({ sort_field: sortField, sort_order: sortOrder }) => ({
+    sort_field: SORT_MAP[sortField] || sortField,
+    sort_order: sortOrder,
+  }));
+};
+
+// TODO (Jiawei): Use node builder instead of strings
+const getFilter = ({ runId, message }: { runId?: string; message?: string }) => {
```

Comment:
I tried to add nodebuilders but decided to wait for now because the `rules_client.getActionErrorLog` function, which calls `eventLogClient.findEventsBySavedObjectIds`, takes filter as a string only. Digging deeper into that function reveals that we eventually call `getQueryBody` which has this code:

```
  try {
    dslFilterQuery = filter ? toElasticsearchQuery(fromKueryExpression(filter)) : undefined;
  } catch (err) {
    logger.debug(
      `esContext: Invalid kuery syntax for the filter (${filter}) error: ${JSON.stringify({
        message: err.message,
        statusCode: err.statusCode,
      })}`
    );
    throw err;
  }
```
Unless there is a way to convert DSL back to KQL (I couldn't find away, would love to know if there's a way), we would have to modify this `event_logger` function to support both the KueryNode and KQL strings.